airbase-policy/src/main/resources/checkstyle/airbase-checks.xml

```diff
@@ -165,10 +165,10 @@
         <module name="ParameterName" />
         <module name="StaticVariableName" />
         <module name="ClassTypeParameterName">
-            <property name="format" value="^[A-Z][0-9]?$" />
+            <property name="format" value="^[A-Z][A-Z0-9]*$" />
         </module>
         <module name="MethodTypeParameterName">
-            <property name="format" value="^[A-Z][0-9]?$" />
+            <property name="format" value="^[A-Z][A-Z0-9]*$" />
```

Comment:
In the past I found it convenient to name my params camel case rather than all-caps (in the case where single-char param names were not enough).
Since we now are moving from single-char (plus optional number, whatever) to multi-char, we can make any choice we want.

Reply:
I think it might be confusing to mix the type variable and normal class name namespace.  I could imagine wanting an underscore in some cases.... I suggest that we expand this more when we find another use case.

BTW my use case was for a between function, I wanted to use `<V, MIN, MAX>`

Reply:
> In the past I found it convenient to name my params camel case rather than all-caps (in the case where single-char param names were not enough).

Yeah, I like doing that, too. For example, I once was working on a library to manipulate graphs, and I used NodeType and EdgeType as my type variables.

Reply:
> I think it might be confusing to mix the type variable and normal class name namespace.

It *is* the same namespace, though. Type variables have precedence over class names, just like local variables have precedence over instance variables. IDEs like IntelliJ will highlight them differently.

Reply:
So it's 2-1 :)

Reply:
@martint yes, it _is_ the same namespace, which is why it can be confusing since you can shadow types.  Anyway, I don't feel strongly.  I just have a preference for relaxing when we need stuff.

Reply:
The IDE does highlight generic parameters differently, though that doesn't help on GitHub, etc. The advantage of all uppercase is that you know visually it can't be a class name (at least one that follows our naming convention).

Reply:
> The IDE does highlight generic parameters differently, though that doesn't help on GitHub, etc. The advantage of all uppercase is that you know visually it can't be a class name (at least one that follows our naming convention).

But that's the same argument we've made for ages against having a naming convention for local variables vs instance methods like many projects do (not that I'm advocating for that!). It's easy to spot them in an IDE, but not in github, etc.